conbench/entities/run.py

```diff
@@ -452,18 +429,41 @@ def commit_hardware_run_map():
     return results
 
 
-class GitHubCreate(marshmallow.Schema):
+class SchemaGitHubCreate(marshmallow.Schema):
+    """
+    GitHub-flavored commit info object
+    """
+
     @marshmallow.pre_load
-    def change_pr_number_empty_string_to_none(self, data, **kwargs):
-        if "pr_number" in data:
-            data["pr_number"] = data["pr_number"] or None
+    def change_empty_string_to_none(self, data, **kwargs):
+        """For the specific situation of empty string being provided,
+        treat this a None, _before_ schema validation.
+
+        This for example alles the client to set pr_number to an empty string
+        and this has the same meaning as setting it to `null` in the JSON doc.
+
+        Otherwise, an empty string results in 'Not a valid integer' (for
+        pr_number, at least).
+        """
+        for k in ("pr_number", "branch"):
+            if data.get(k) == "":
+                data[k] = None
 
         return data
 
     commit = marshmallow.fields.String(
         required=True,
         metadata={
-            "description": "The 40-character commit hash of the repo being benchmarked"
+            "description": conbench.util.dedent_rejoin(
+                """
+                The commit hash of the benchmarked code.
```

Comment:
Can we still specify that it should be full (40-char) here, instead of a shortened version? There's so much software out there that accepts a shortened version, and we should not accept that

Reply:
Yes! Adding a commit:

Adding this back to the textual specification, and also adding a new validation error.

Reply:
Backing out of this. Have been working for 40 minutes to unbreak our tests, unsuccessfully. The test suite relies on being able to use (very) short commit hashes. Pushing my WIP branch to https://github.com/conbench/conbench/pull/1136.

Reply:
Oof, I forgot about that...